apps/main/src/dex/hooks/useAutoRefresh.tsx

```diff
@@ -1,36 +1,34 @@
-import { useCallback } from 'react'
+import { useCallback, useMemo } from 'react'
 import curvejsApi from '@/dex/lib/curvejs'
 import useStore from '@/dex/store/useStore'
-import type { NetworkDef } from '@ui/utils'
 import { type CurveApi, useConnection } from '@ui-kit/features/connect-wallet'
 import usePageVisibleInterval from '@ui-kit/hooks/usePageVisibleInterval'
 import { REFRESH_INTERVAL } from '@ui-kit/lib/model'
 import { useGasInfoAndUpdateLib } from '@ui-kit/lib/model/entities/gas-info'
-import { useNetworkByChain, useNetworks } from '../entities/networks'
+import { useNetworks } from '../entities/networks'
 
-export const useAutoRefresh = (networkDef: NetworkDef) => {
+export const useAutoRefresh = (chainId: number | undefined) => {
   const { curveApi } = useConnection()
-
+  const { data: networks } = useNetworks()
   const fetchPools = useStore((state) => state.pools.fetchPools)
-  const poolDataMapper = useStore((state) => state.pools.poolsMapper[networkDef.chainId])
+  const poolDataMapper = useStore((state) => chainId && state.pools.poolsMapper[chainId])
   const fetchPoolsVolume = useStore((state) => state.pools.fetchPoolsVolume)
   const fetchPoolsTvl = useStore((state) => state.pools.fetchPoolsTvl)
   const setTokensMapper = useStore((state) => state.tokens.setTokensMapper)
   const fetchAllStoredBalances = useStore((state) => state.userBalances.fetchAllStoredBalances)
 
-  const { data: networks } = useNetworks()
-  const { data: network } = useNetworkByChain({ chainId: networkDef.chainId })
+  const network = useMemo(() => chainId && networks[chainId], [chainId, networks])
```

Comment:
But this is pretty much what `useNetworkByChain` does, not sure why you inlined it. Not wrong, just odd